python-frontend/src/main/java/org/sonar/python/tree/CallExpressionImpl.java

```diff
@@ -187,13 +188,23 @@ public List<Expression> typeDependencies() {
 
   @Override
   public PythonType typeV2() {
-    if (callee().typeV2() instanceof ClassType classType) {
-      return new ObjectType(classType);
+    TypeSource typeSource = computeTypeSource();
+    PythonType pythonType = returnTypeOfCall(callee().typeV2());
+    return pythonType != PythonType.UNKNOWN ? new ObjectType(pythonType, typeSource) : PythonType.UNKNOWN;
+  }
+
+  static PythonType returnTypeOfCall(PythonType calleeType) {
+    if (calleeType instanceof ClassType classType) {
+      return classType;
```

Comment:
should it be an ObjectType of a ClassType?

Reply:
The actual `ObjectType` is only created when its `TypeSource` is known. In `typeV2()`, I get both the type and the source from different methods, then instantiate the `ObjectType` with that information.